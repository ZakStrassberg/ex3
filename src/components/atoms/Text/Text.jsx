import React from 'react';
import { noop } from 'lodash';
import PropTypes from 'prop-types';

const Text = ({ value, name, onChange }) => (
  <input type="text" value={value} name={name} onChange={onChange} />
);

Text.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string,
  onChange: PropTypes.func,
};

Text.defaultProps = {
  name: 'textbox',
  onChange: noop,
};

export default Text;
