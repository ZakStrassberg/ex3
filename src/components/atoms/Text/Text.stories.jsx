import React from 'react';

import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import Text from './';

storiesOf('Text', module)
  .add('Empty', () => <Text value="" />)
  .add('With value', () => <Text value="Text input" />);
