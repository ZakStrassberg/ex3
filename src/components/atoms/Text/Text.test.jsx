import React from 'react';

import Text from './';

describe('Checkbox', () => {
  it('should shallow render', () => {
    const wrapper = shallow(<Text value="" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render value when passed props', () => {
    const value = 'Test';
    const wrapper = shallow(<Text value={value} />);
    expect(wrapper.find('input').props().value).toBe(value);
    expect(wrapper).toMatchSnapshot();
  });
});
