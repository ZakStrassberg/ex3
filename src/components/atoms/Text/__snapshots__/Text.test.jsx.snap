// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should shallow render 1`] = `
<input
  name="textbox"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
